concordium-consensus/haskell-bench/LRUCache.hs

```diff
@@ -0,0 +1,83 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Main where
+
```

Comment:
Can you perhaps put some module level documentation on what this benchmark does and how. 
(You already said it a few times now, but it will serve us nicely in the future)

Reply:
done